packages/flame/lib/src/particles/particle.dart

```diff
@@ -37,115 +39,114 @@ abstract class Particle {
+  void render(Canvas canvas) {}
 
-  /// Updates internal [Timer] of this [Particle]
-  /// which defines its position on the lifespan.
-  /// Marks [Particle] for removal when it is over.
+  /// Updates the [_timer] of this [Particle].
   void update(double dt) {
     _timer?.update(dt);
   }
 
-  /// A control method allowing a parent of this [Particle]
-  /// to pass down it's lifespan. Allows to only specify desired lifespan
-  /// once, at the very top of the [Particle] tree which
-  /// then will be propagated down using this method.
-  /// See SingleChildParticle or [ComposedParticle] for details.
+  /// A control method allowing a parent of this [Particle] to pass down it's
+  /// lifespan.
+  ///
+  /// Allows to only specify desired lifespan once, at the very top of the
+  /// [Particle] tree which then will be propagated down using this method.
+  ///
+  /// See `SingleChildParticle` or [ComposedParticle] for details.
   void setLifespan(double lifespan) {
+    // TODO: Maybe make it into a setter/getter?
     _lifespan = lifespan;
     _timer?.stop();
-    void removeCallback() => _shouldBeRemoved = true;
-    _timer = Timer(lifespan, callback: removeCallback);
-    _timer!.start();
+    _timer = Timer(lifespan, callback: () => _shouldBeRemoved = true)..start();
   }
 
-  /// Wraps this particle with [TranslatedParticle]
-  /// statically repositioning it for the time
-  /// of the lifespan.
-  Particle translated(Offset offset) {
+  /// Wraps this particle with a [TranslatedParticle].
+  ///
+  /// Statically repositioning it for the time of the lifespan.
+  Particle translated(Vector2 offset) {
     return TranslatedParticle(
       offset: offset,
       child: this,
       lifespan: _lifespan,
     );
   }
 
-  /// Wraps this particle with [MovingParticle]
-  /// allowing it to move from one [Offset]
-  /// on the canvas to another one.
+  /// Wraps this particle with a [MovingParticle].
+  ///
+  /// Allowing it to move from one [Vector2] to another one.
   Particle moving({
-    Offset from = Offset.zero,
-    required Offset to,
+    Vector2? from,
+    required Vector2 to,
     Curve curve = Curves.linear,
   }) {
     return MovingParticle(
-      from: from,
+      from: from ?? Vector2.zero(),
       to: to,
       curve: curve,
       child: this,
       lifespan: _lifespan,
     );
   }
 
-  /// Wraps this particle with [AcceleratedParticle]
-  /// allowing to specify desired position speed and acceleration
-  /// and leave the basic physics do the rest.
+  /// Wraps this particle with a [AcceleratedParticle].
+  ///
+  /// Allowing to specify desired position speed and acceleration and leave
+  /// the basic physics do the rest.
   Particle accelerated({
-    Offset acceleration = Offset.zero,
-    Offset position = Offset.zero,
-    Offset speed = Offset.zero,
+    Vector2?
+        acceleration, // TODO: Any of these should maybe be required? What else is the use
```

Comment:
Since they have defaults, makes sense

Reply:
The question is, which one. I would say acceleration.

Reply:
definitely acceleration, no sense in having an AccParticle if the acc is 0